crates/ruff_linter/src/rules/ruff/rules/ambiguous_unicode_character.rs

```diff
@@ -171,16 +173,59 @@ impl Violation for AmbiguousUnicodeCharacterComment {
     }
 }
 
-/// RUF001, RUF002, RUF003
-pub(crate) fn ambiguous_unicode_character(
+/// RUF003
+pub(crate) fn ambiguous_unicode_character_comment(
     diagnostics: &mut Vec<Diagnostic>,
     locator: &Locator,
     range: TextRange,
-    context: Context,
     settings: &LinterSettings,
 ) {
     let text = locator.slice(range);
+    ambiguous_unicode_character(diagnostics, text, range, Context::Comment, settings);
+}
 
+/// RUF001, RUF002
+pub(crate) fn ambiguous_unicode_character_string(checker: &mut Checker, string_like: StringLike) {
+    let context = if checker.semantic().in_docstring() {
+        Context::Docstring
+    } else {
+        Context::String
+    };
+
+    match string_like {
+        StringLike::StringLiteral(string_literal) => {
```

Comment:
Nit: I think you can simplify this by early returning if it is a byte literal and you can then operate directly on the string like (no need for the branching)

Reply:
Hmm, yeah. I could probably use `chain` and combine it into a single call (probably not what you meant).

Reply:
Or do you mean to take the range of the entire `string_like`? But that would include implicitly concatenated strings.

Reply:
I added another suggestion with the code changes. I don't understand the remark on how it would include implicitly concatenated strings since we're operating on string literal's.

Reply:
> I don't understand the remark on how it would include implicitly concatenated strings since we're operating on string literal's.

Because 😅:

```rust
pub enum StringLike<'a> {
    StringLiteral(&'a ast::ExprStringLiteral),
    BytesLiteral(&'a ast::ExprBytesLiteral),
    FStringLiteral(&'a ast::FStringLiteralElement),
}
```

Reply:
> Argh, is it the whole string literal confusion all over again.

Yeah, sorry 😓. I'll try to sort out at least the name confusion over the weekend.